src/newgrf_house.cpp

```diff
@@ -492,7 +492,7 @@ static uint32_t GetDistanceFromNearbyHouse(uint8_t parameter, TileIndex tile, Ho
 				local_houseid = nearby_house_id;
 			} else {
 				local_houseid = (hs->grf_prop.grffile == this->ro.grffile ? 1 : 2) << 8;
-				local_houseid |= hs->grf_prop.local_id;
+				local_houseid |= ClampTo<uint8_t>(hs->grf_prop.local_id);
```

Comment:
What's the most elegant way to break this, I wonder? Looking for a specific HouseID at a specific tile probably isn't a common usecase (I've done some silly stuff in ITL Houses, and not touched this one), but I imagine a false negative is a better failure state than a false positive. In other words, instead of returning an incorrect ID that might mean something to the author (as it's in their house GRF), would it better to return the fallback base game house ID, that they are not looking for at all?

At the very least, I think a comment for future-us would be helpful.

Reply:
Probably just adding a new variable, like what happened for stations.

Reply:
Do you want to do that here or as a later PR?

If later, I think a comment here noting that this ID can't read IDs above 255, and a similar note in the GRF specs, would be enough and I'd be happy to approve. 🙂

Reply:
Later PR I think. Comment added here for now.